cli-quarkus/src/main/java/org/eclipse/hono/cli/adapter/amqp/AmqpAdapter.java

```diff
@@ -0,0 +1,610 @@
+                            "The identifier of the device to stop receiving commands for.",
+                            """
+                            If not set explicitly, the identifier of the device that has authenticated to the AMQP \
+                            adapter will be used.
+                            """,
+                            """
+                            Authenticated gateway devices can use this parameter to stop receiving commands for another \
+                            device that the gateway is authorized to act on behalf of.
+                            """,
+                            """
+                            Unauthenticated clients must provide a non-{@code null} value to indicate the device to \
+                            stop receiving commands for.
+                            """,
+                            CommandUtils.DESCRIPTION_ENV_VARS
+                    },
+                    order = 21,
+                    converter = StringVariableConverter.class)
+            final String deviceId) {
+        Optional.ofNullable(activeConsumers.remove(getConsumerKey(tenantId, deviceId)))
+            .ifPresent(consumer -> consumer.close(null));
+    }
+
+    @CommandLine.Command(
+            name = "telemetry",
+            description = {"Send a telemetry message."},
+            mixinStandardHelpOptions = true,
+            subcommands = CommandLine.HelpCommand.class,
+            versionProvider = PropertiesVersionProvider.class,
+            sortOptions = false)
+    void sendTelemetry(
+            @CommandLine.Mixin
+            final TelemetrySendingOptions options) {
+
+        checkDeviceSpec(options.tenantId, options.deviceId);
+        getClient()
+            .compose(client -> client.sendTelemetry(
+                    QoS.AT_MOST_ONCE,
+                    Optional.ofNullable(options.payload).map(Buffer::buffer).orElse(null),
+                    options.contentType,
+                    options.tenantId,
+                    options.deviceId,
+                    null))
+            .onFailure(t -> {
+                System.err.println("Cannot send telemetry message.");
+                handleUnauthorizedError(t, options.deviceId);
+            })
+            .toCompletionStage()
+            .toCompletableFuture()
+            .join();
+    }
+
+    @CommandLine.Command(
+            name = "event",
+            description = {"Send an event message."},
+            mixinStandardHelpOptions = true,
+            subcommands = CommandLine.HelpCommand.class,
+            versionProvider = PropertiesVersionProvider.class,
+            sortOptions = false)
+    void sendEvent(
+            @CommandLine.Mixin
+            final TelemetrySendingOptions options) {
+
+        checkDeviceSpec(options.tenantId, options.deviceId);
+        getClient()
+            .compose(f -> f.sendEvent(
+                    Optional.ofNullable(options.payload).map(Buffer::buffer).orElse(null),
```

Comment:
IMHO this is first of all a bug in the AMQP device client because the AMQP adapter actually supports sending empty messages.
However, I agree that the command line should allow sending an empty notification simply by omitting the payload option.